controllers/usersignup/usersignup_controller.go

```diff
@@ -194,7 +198,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, r.updateStatus(logger, userSignup, r.setStatusDeactivated)
 	}
 
-	return reconcile.Result{}, r.ensureNewMurIfApproved(logger, config, userSignup)
+	if err := r.ensureNewMurIfApproved(logger, config, userSignup); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, err
```

Comment:
we could use the original format, right?
```suggestion
	return reconcile.Result{}, r.ensureNewMurIfApproved(logger, config, userSignup)
```